### PR TITLE
[WFLY-11179]:  Messaging http-acceptor should expose a capability and use it to control dependencies.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding a capability and a requirement on http-upgrade-registry

JIRA: https://issues.jboss.org/browse/WFLY-11179

### CODE REVIEW

**Comment by kabir** [comment]
Should be fine, if CI passes

**Comment by ehsavoie** [comment]
requires core upgrade

**Comment by jamezp** [comment]
retest this please

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPAcceptorDefinition.java [hunk 1]
```diff
@@ -57,21 +61,25 @@
             .setRequired(false)
             .setAllowExpression(true)
             .build();
-
+    public static final RuntimeCapability<Void> CAPABILITY = RuntimeCapability.Builder.of("org.wildfly.messaging.activemq", true, HTTPUpgradeService.class)
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPAcceptorDefinition.java]: This can be package protected.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPAcceptorDefinition.java (hunk 1 above)
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPAcceptorDefinition.java]: Fixed

**Review by bstansberry — APPROVED** [review]

**Pull request merged**